events/event_queue.go

```diff
@@ -195,6 +197,11 @@ func (t SyncEventConsumer) EventConsumer() *eventconsumer.EventConsumer {
 }
 
 func (t *SyncEventConsumer) Handle(ctx api.Context) (int, error) {
+	tracer := otel.GetTracerProvider().Tracer("event-tracer")
+	traceCtx, span := tracer.Start(ctx, "event-queue-"+strings.Join(t.watchEvents, "/"))
+	ctx = ctx.WithContext(traceCtx)
+	defer span.End()
+
```

Comment:
I think this could be 

```go
ctx, span := ctx.Span(tracer, "event-queue-"+strings.Join(t.watchEvents, "/"))
defer span.End()

```
Where `tracer := otel.GetTracerProvider().Tracer("event-tracer")` is a package level variable.

Reply:
Makes sense, added a `ctx.StartTrace()` function